cilium-cli/connectivity/tests/errors.go

```diff
@@ -157,10 +157,16 @@ func (n *noErrorsInLogs) Run(ctx context.Context, t *check.Test) {
 				// Do not check for container restarts for Cilium v1.16 and earlier.
 				ignore := n.ciliumVersion.LT(semver.MustParse("1.17.0"))
 
-				// Ignore Cilium operator restarts for the moment, as they can
-				// legitimately happen in case it loses the leader election due
-				// to temporary control plane blips.
-				ignore = ignore || container == "cilium-operator"
+				// Check for legitimate cilium-operator restart patterns
+				if container == "cilium-operator" && restarts > 0 {
+					var logs bytes.Buffer
+					err := client.GetLogs(ctx, pod.Namespace, pod.Name, container, opts, &logs)
+					if err == nil {
```

Comment:
What if the err is not nil? Presumably the error should be logged somewhere so that it doesn't silently fail?